R/utils.R

```diff
@@ -46,3 +46,26 @@ include_parent_datanames <- function(dataname, join_keys) {
 
   return(unique(c(parents, dataname)))
 }
+
+
+
+#' Create a `FilteredData`
+#'
+#' Create a `FilteredData` object from a `teal_data` object
+#' @param x (`teal_data`) object
+#' @return (`FilteredData`) object
+#' @keywords internal
+teal_data_to_filtered_data <- function(x) { #     nolint
+  checkmate::assert_class(x, "teal_data")
+  datanames <- x@datanames
```

Comment:
`datanames` is only used once and in a simple call, consider using `x@datanames` directly.

Reply:
yes - please use getters from teal.data API instead of S4 accessors to loosen coupling to its implementation